src/test/java/com/amazonaws/encryptionsdk/jce/JceMasterKeyTest.java

```diff
@@ -0,0 +1,70 @@
+package com.amazonaws.encryptionsdk.jce;
+
+import org.junit.Test;
+
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.*;
+
+public class JceMasterKeyTest {
+
+    private static final SecretKey SECRET_KEY = new SecretKeySpec(new byte[1], "AES");
+    private static final PrivateKey PRIVATE_KEY;
+    private static final PublicKey PUBLIC_KEY;
+
+    static {
+        try {
+            KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+            KeyPair keyPair = keyPairGenerator.generateKeyPair();
+            PUBLIC_KEY = keyPair.getPublic();
+            PRIVATE_KEY = keyPair.getPrivate();
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private JceMasterKey jceGetInstance(final String algorithmName) {
+        return JceMasterKey.getInstance(SECRET_KEY, "mockProvider", "mockKey", algorithmName);
+    }
+
+    private JceMasterKey jceGetInstanceAsymmetric(final String algorithmName) {
+        return JceMasterKey.getInstance(PUBLIC_KEY, PRIVATE_KEY, "mockProvider",  "mockKey",
+                algorithmName);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testGetInstanceInvalidWrappingAlgorithm() {
+        jceGetInstance("blatently/unsupported/algorithm");
+    }
+
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testGetInstanceAsymmetricInvalidWrappingAlgorithm() {
+        jceGetInstanceAsymmetric("rsa/ec/unsupportedAlgorithm");
+    }
+
+    /**
+     * Calls JceMasterKey.getInstance with differently cased wrappingAlgorithm names.
+     * Passes if no Exception is thrown.
+     * Relies on passing an invalid algorithm name to result in an Exception.
+     */
+    @Test
+    public void testGetInstanceCase1() {
```

Comment:
NP: Bytes are cheap. Human cognition trying to determine what a test is doing from a vague name is not.

Reply:
Added more descriptive names.